### ISSUE TITLE
modified count is higher compared to bulkWrite with unmodified items

### ISSUE THREAD

**Issue by jschuur** [meanie/mongoose-upsert-many#13]
I've noticed that nModified shows all items were modified even if I send in the exact same data twice in a row on an uninitialised collection. When I do the same with a `bulkWrite`, this doesn't seem to be the case. See the example and output below.

Isn't this a sign that much more database activity occurred than it should have for unmodified data? In my scenario, I intend to call this with a few hundred or even thousand items, so I'm not sure if the lengthier `bulkWrite` code isn't more performant.

This is using Mongoose 5.9.5 and 1.5.2 of your plugin.

How many round trips to the database does this make BTW? Will this make one call per item in the array passed in?

```
import mongoose from 'mongoose';
import upsertMany from '@meanie/mongoose-upsert-many';

mongoose.plugin(upsertMany);

const Schema = mongoose.Schema;

const Thing = mongoose.model('Thing', new Schema({
  name: String,
  count: Number
}, {
  versionKey: false,
  upsertMatchFields: ['name']
}));

const things = [
  { name: 'couch', count: 1 },
  { name: 'table', count: 2 },
  { name: 'chair', count: 4 }
];

(async () => {
  var db = await mongoose.connect(process.env.MONGODB_URL, {
    useNewUrlParser: true,
    useUnifiedTopology: true });

  var results = await Thing.upsertMany(things);
  console.log(`Initial write: ${JSON.stringify(results, null, 2)}`);

  results = await Thing.upsertMany(things);
  console.log(`Update with upsertMany: ${JSON.stringify(results, null, 2)}`);

  results = await Thing.bulkWrite(things.map((thing)  => ({
    updateOne: {
      filter: { name: thing.name },
      update: thing,
      upsert: true
    }
  })));
  console.log(`Update with bulkWrite: ${JSON.stringify(results, null, 2)}`);

  mongoose.disconnect();
})();
```

```
Initial write: {
  "ok": 1,
  "writeErrors": [],
  "writeConcernErrors": [],
  "insertedIds": [],
  "nInserted": 0,
  "nUpserted": 3,
  "nMatched": 0,
  "nModified": 0,
  "nRemoved": 0,
  "upserted": [
    {
      "index": 0,
      "_id": "5e8efaacdfdfb7d5cd092076"
    },
    {
      "index": 1,
      "_id": "5e8efaacdfdfb7d5cd092077"
    },
    {
      "index": 2,
      "_id": "5e8efaacdfdfb7d5cd092078"
    }
  ]
}
Update with upsertMany: {
  "ok": 1,
  "writeErrors": [],
  "writeConcernErrors": [],
  "insertedIds": [],
  "nInserted": 0,
  "nUpserted": 0,
  "nMatched": 3,
  "nModified": 3,
  "nRemoved": 0,
  "upserted": []
}
Update with bulkWrite: {
  "ok": 1,
  "writeErrors": [],
  "writeConcernErrors": [],
  "insertedIds": [],
  "nInserted": 0,
  "nUpserted": 0,
  "nMatched": 3,
  "nModified": 0,
  "nRemoved": 0,
  "upserted": []
}
```

**Comment by adamreisnz** [comment]
This plugin uses `initializeUnorderedBulkOp` under the hood. It will add a new bulk op for each item, and then execute it all in one single call. So afaik it's just one round trip database call.

However, I will try to update this plugin to use the new `bulkWrite` method instead, that might be the reason for the discrepancy.

**Comment by adamreisnz** [comment]
I have released a v2.0.0 which allows for some more flexibility and additional configuration. See updated readme for usage.

This will now use `updateOne` by default, instead of `replaceOne`, resulting in the same modified count:

```
Initial write: {
  "ok": 1,
  "writeErrors": [],
  "writeConcernErrors": [],
  "insertedIds": [],
  "nInserted": 0,
  "nUpserted": 0,
  "nMatched": 3,
  "nModified": 0,
  "nRemoved": 0,
  "upserted": []
}
Update with upsertMany: {
  "ok": 1,
  "writeErrors": [],
  "writeConcernErrors": [],
  "insertedIds": [],
  "nInserted": 0,
  "nUpserted": 0,
  "nMatched": 3,
  "nModified": 0,
  "nRemoved": 0,
  "upserted": []
}
Update with bulkWrite: {
  "ok": 1,
  "writeErrors": [],
  "writeConcernErrors": [],
  "insertedIds": [],
  "nInserted": 0,
  "nUpserted": 0,
  "nMatched": 3,
  "nModified": 0,
  "nRemoved": 0,
  "upserted": []
}
```
